base/wrappers/data_access.go

```diff
@@ -0,0 +1,212 @@
+package wrappers
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/raito-io/cli/base/access_provider"
+	"github.com/raito-io/cli/base/access_provider/sync_from_target"
+	"github.com/raito-io/cli/base/access_provider/sync_to_target"
+	"github.com/raito-io/cli/base/access_provider/sync_to_target/naming_hint"
+	"github.com/raito-io/cli/base/util/config"
+	e "github.com/raito-io/cli/base/util/error"
+)
+
+//go:generate go run github.com/vektra/mockery/v2 --name=AccessProviderHandler --with-expecter
+type AccessProviderHandler interface {
+	AddAccessProviders(dataAccessList ...*sync_from_target.AccessProvider) error
+}
+
+//go:generate go run github.com/vektra/mockery/v2 --name=AccessProviderFeedbackHandler --with-expecter
+type AccessProviderFeedbackHandler interface {
+	AddAccessProviderFeedback(accessProviderId string, accessFeedback ...sync_to_target.AccessSyncFeedbackInformation) error
+}
+
+//go:generate go run github.com/vektra/mockery/v2 --name=AccessProviderSyncer --with-expecter --inpackage
+type AccessProviderSyncer interface {
+	SyncAccessProviderFromTarget(ctx context.Context, accessProviderHandler AccessProviderHandler, configMap *config.ConfigMap) error
+	SyncAccessProviderToTarget(ctx context.Context, rolesToRemove []string, accesses map[string]sync_to_target.EnrichedAccess, feedbackHandler AccessProviderFeedbackHandler, configMap *config.ConfigMap) error
```

Comment:
I would not use 'accesses' ... not really the right plural I think? Just use 'access'
Same below